Formula/h/hackernews-tui.rb

```diff
@@ -0,0 +1,23 @@
+class HackernewsTui < Formula
+  desc "Terminal UI to browse Hacker News"
+  homepage "https://github.com/aome510/hackernews-TUI"
+  url "https://github.com/aome510/hackernews-TUI/archive/refs/tags/v0.13.4.tar.gz"
+  sha256 "77abc579f58ed33905f00eb4d6047b1d7f30a0b4ed98b44b36ff95648c6ebae0"
+  license "MIT"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "build", "--manifest-path", "hackernews_tui/Cargo.toml", "--release"
+    bin.install "target/release/hackernews_tui" => "hackernews-tui"
+  end
+
+  test do
+    system "#{bin}/hackernews-tui", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?
Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I did read through that section of the documentation before submitting, however, I am not sure how to design a test when none of the app's functionality is available in a headless manner.
I therefore leaned on this &mdash;
> For example foo build-foo input.foo is a good test and (despite their widespread use) foo --version and foo --help are bad tests. **However, a bad test is better than no test at all.**